tsdb/query_executor.go

```diff
@@ -140,7 +151,12 @@ func (q *QueryExecutor) ExecuteQuery(query *influxql.Query, database string, chu
 	results := make(chan *influxql.Result)
 
 	go func() {
-		defer close(results)
+		q.statMap.Add(statQueriesActive, 1)
+		defer func(start time.Time) {
+			close(results)
```

Comment:
Not sure why the `close` has moved into here? I think the following would be fine:

```go
defer close(results)
defer func(start time.Time) { ... }(time.Now())
```